docs/sourcemaps.asciidoc

```diff
@@ -1,38 +1,192 @@
+module.exports = {
+  entry: 'app.js',
+  output: {
+    filename: 'app.min.js',
+    path: './dist'
+  },
+  devtool: 'source-map',
+  plugins: [
+    new webpack.DefinePlugin({'serviceVersion': JSON.stringify(serviceVersion)}),
+    new TerserPlugin({
+      sourceMap: true
+    })
+  ]
+}
+----
+<1> If you're using a different method of defining `serviceVersion`, you can set it here.
+
+[float]
+[[sourcemap-rum-upload]]
+=== Upload the source map to APM Server
+
+The APM Server exposes a <<sourcemap-api,source map endpoint>> for uploading source maps.
 Source maps are cached in memory for as long as the <<rum-sourcemap-cache,cache expiration>> setting indicates.
+Source maps must be attached as a `file upload` and you can configure an API key or secret token to restrict uploads.
+
+Before uploading a source map, ensure that:
+
+1. RUM support is enabled in APM Server by setting <<rum-enable,`apm-server.rum.enabled`>> to `true`.
+2. <<config-sourcemapping-enabled,`apm-server.rum.source_mapping.enabled`>> is `true` (the default).
+This means that if a source map has previously been uploaded,
+source mapping will automatically be applied to all incoming transactions and errors.
+
+Let's look at two different ways to upload a source map: curl and a custom application.
+Each example includes the four fields necessary for APM Server to later map minified code to its source:
+
+* `service_name` - Should match the `serviceName` from step one
+* `service_version` - Should match the `serviceVersion` from step one
+* `bundle_filepath` - The absolute path of the final bundle as used in the web application
+* `sourcemap` - The location of the source map.
+If you have multiple source maps, you'll need to upload each individually.
 
-[[sourcemap-apply]]
 [float]
-== How source maps are applied
+[[sourcemap-curl]]
+==== Upload via curl
+
+Here’s an example curl request that uploads the source map file created in the previous step:
 
-APM Server needs to find the correct source map for every `stack trace frame` in an event.
-To do so, it tries the following:
+[source,console]
+----
+SERVICEVERSION=`node -e "console.log(require('./package.json').version);"` && \ <1>
+curl http://localhost:8200/assets/v1/sourcemaps -X POST \
+    -F sourcemap="@./dist/app.min.js.map" \
+    -F service_version="$SERVICEVERSION" \
+    -F bundle_filepath="http://localhost/app.min.js" \
+    -F service_name="myService"
+----
+<1> This example uses the version from `package.json`.
 
-* Compare the event's `service.name` with the source map's `service_name`
-* Compare the event's `service.version` with the source map's `service_version`
-* Compare the stack trace frame's `abs_path` with the source map's `bundle_filepath`
+Add a <<secret-token,secret token>> to your curl command using an Authorization header:
 
-If a source map is found, the following attributes in the `stack trace frames` are overwritten:
+[source,console]
+----
+-H "Authorization: Bearer <token>"
```

Comment:
The "copy as curl: is not so useful when splitting the command like this, maybe include this header in the full curl example above, and mention the ApiKey alternative?